libs/wire-api/src/Wire/API/Team/Conversation.hs

```diff
@@ -36,76 +36,143 @@ module Wire.API.Team.Conversation
   )
 where
 
-import Control.Lens (At (at), makeLenses, over, (?~))
-import Data.Aeson hiding (fieldLabelModifier)
+import Control.Lens (makeLenses, (?~))
+import qualified Data.Aeson as A
 import Data.Id (ConvId)
-import Data.Proxy
-import Data.Swagger
+import Data.Schema
+import qualified Data.Swagger as S
 import qualified Data.Swagger.Build.Api as Doc
 import Imports
+import Wire.API.Routes.Version
 import Wire.Arbitrary (Arbitrary, GenericUniform (..))
 
 --------------------------------------------------------------------------------
 -- TeamConversation
 
-newtype TeamConversation = TeamConversation
+-- FUTUREWORK: The version tag type argument can be dropped as soon as versions
+-- below V2 are not supported any more.
+newtype TeamConversation (v :: *) = TeamConversation
```

Comment:
```suggestion
-- Derivation of various JSON instances is verbose with this version tag type variable, but this can be simplified once the old version can be dropped.
newtype TeamConversation (v :: *) = TeamConversation
```